drracket-tool-lib/drracket/private/standalone-module-browser.rkt

```diff
@@ -365,143 +363,137 @@
-               [parent search-hp]
-               [callback
-                (λ (tf evt)
-                  (define val (send tf get-value))
-                  (define reg (and (not (string=? val ""))
-                                   (regexp (regexp-quote (send tf get-value)))))
-                  (update-found-and-search-hits reg))]))
-        (define search-hits (new message% [parent search-hp] [label ""] [auto-resize #t]))
-        (define (update-found-and-search-hits reg)
-          (send pasteboard begin-edit-sequence)
-          (define count 0)
-          (define phases (set))
-          (let loop ([snip (send pasteboard find-first-snip)])
-            (when snip
-              (when (is-a? snip boxed-word-snip<%>)
-                (define fn (send snip get-filename))
-                (define found? 
-                  (and reg fn (regexp-match reg (path->string fn))))
-                (when (or (not reg) found?) 
-                  (for ([phase (in-list (send snip get-require-phases))])
-                    (set! phases (set-add phases phase)))
-                  (set! count (+ count 1)))
-                (send snip set-found! found?))
-              (loop (send snip next))))
-          
-          (send search-hits set-label 
-                (string-append
-                 (if reg
-                     (format "~a found" count)
-                     (format "~a total" count))
-                 (render-phases phases)))
-          (send pasteboard end-edit-sequence))
-        (update-found-and-search-hits #f) ;; only to initialize search-hits
-        
-        (set! update-label
-              (λ (s)
-                (if (and s (not (null? s)))
-                    (let* ([currently-over (car s)]
-                           [fn (send currently-over get-filename)]
-                           [lines (send currently-over get-lines)])
-                      (when (and fn lines)
-                        (define label (format filename-constant fn lines))
-                        (define pkg (send currently-over get-pkg))
-                        (when pkg (set! label (string-append (format pkg-constant pkg) "  " label)))
-                        (send label-message set-label label)))
-                    (send label-message set-label ""))))
-        
-        (send pasteboard set-name-length 
-              (case (preferences:get 'drracket:module-browser:name-length)
-                [(0) 'long]
-                [(1) 'long]
-                [(2) 'long]
-                [(3) 'very-long]))
-        ;; shouldn't be necessary here -- need to find callback on editor
-        (send pasteboard render-snips)
-        
-        (send frame show #t)))))
+  (define success? (fill-pasteboard pasteboard filename show-status void))
+  (kill-thread thd)
+  (parameterize ([current-eventspace progress-eventspace])
+    (queue-callback (λ () (send progress-frame show #f))))
+  (when success?
+    (let ()
```

Comment:
This let was in the original, but shouldn't it have gone away?

Reply:
Yup. I just checked using `#lang resyntax/test` that Resyntax can refactor this code, so I think the only reason it didn't happen was the fix limit.